docs/source/cc_launcher.md

```diff
@@ -0,0 +1,301 @@
+> builders:
+> - LD_LIBRARY_PATH
+> - LIBPATH
+> - PATH
+> - TMPDIR
+
+When an `externalBuilder` configuration is present, the peer will iterate over
+the list of builders in the order provided, invoking `bin/detect` until one
+completes successfully. If no builder completes `detect` successfully, the
+peer will fallback to using the legacy Docker build process implemented within
+the peer. This means that external builders are completely optional.
+
+In the example above, the peer will attempt to use "my-golang-builder",
+followed by "noop-builder", and finally the peer internal build process.
+
+# Chaincode Packages
+
+As part of the new lifecycle introduced with Fabric 2.0, the chaincode package
+format changed from serialized protocol buffer messages to a gzip compressed
+POSIX tape archive. Chaincode packages created with `peer lifecycle chaincode
+package` use this new format.
+
+## Lifecycle Chaincode Package Contents
+
+A lifecycle chaincode package contains two files. The first file,
+`code.tar.gz` is a gzip compressed POSIX tape archive. This file includes the
+source artifacts for chaincode. Packages created by the peer CLI will place
+the chaincode implementation source under the `src` directory and chaincode
+metadata (like CouchDB indexes) under the `META-INF` directory.
+
+The second file, `metadata.json` is a JSON document with three keys:
+- `type`: the chaincode type (e.g. GOLANG, JAVA, NODE)
+- `path`: for go chaincode, the GOPATH or GOMOD relative path to the main
+  chaincode package; undefined for other types
+- `label`: the chaincode label
+
+## Chaincode Packages and External Builders
+
+When a chaincode package is installed to a peer, the contents of `code.tar.gz`
+and `metadata.json` are not processed prior to calling external builders. This
+affords users a great deal of flexibility in how they package source and
+metadata that will be processed by external builders and launchers.
+
+For example, a custom chaincode package could be constructed that contains a
+pre-compiled, implementation of chaincode in `code.tar.gz` with a
+`metadata.json` that allows a _binary buildpack_ to detect the custom package,
+validate the hash of the binary, and run the program as chaincode.
+
+Another example would be a chaincode package that only contains state database
+index definitions and the data necessary for an external launcher to connect
+to a running chaincode server. In this case, the `build` process would simply
+extract the metadata from the process and `release` would present it to the
+peer.
+
+The only requirements are that `code.tar.gz` can only contain regular file and
+directory entries, and that the entries cannot contain paths that would result
+in files being written outside of the logical root of the chaincode package.
+
+If no configured external builder claims a chaincode package, the peer will
+attempt to process the package as if it were created with the peer CLI.
```

Comment:
the standard Fabric packaging tools (note, the peer CLI, node SDK, and others can create these packagres)

Reply:
I changed this to 
If no configured external builder claims a chaincode package, the peer will attempt to process the package as if it were created with the standard Fabric packaging tools such as the peer CLI or node SDK.